packages/jsii-rosetta/lib/tablets/tablets.ts

```diff
@@ -147,12 +165,21 @@ export class LanguageTablet {
     return Object.values(this.snippets);
   }
 
+  /**
+   * Saves the tablet schema to a file. If the filename has a '.gz' extension the tablet
+   * schema will be compressed, otherwise the tablet schema will be written in raw json.
+   */
   public async save(filename: string) {
     await fs.mkdirp(path.dirname(filename));
-    await fs.writeJson(filename, this.toSchema(), {
-      encoding: 'utf-8',
-      spaces: 2,
-    });
+
+    if (path.extname(filename) === '.gz') {
```

Comment:
hoping to get away with this contract -- supplying a file extension of `.gz` indicates that one is expecting a gzipped file. i feel like this is logical enough to me, but my spidey senses are tingling since the file extension is governing behavior, and we may not want that.

Reply:
Yeah I'd rather it accepts a `compress` property and acts accordingly.

Additionally, you can factor the JSON-ification so it is done once. You can use `Buffer.from` to obtain a `Buffer` from a `string`... So you can stringify in a buffer, then compress that buffer if needed, then write the buffer to disk.

Reply:
(I think) I've done what you're saying here. The side effect is that we no longer get the pretty printed JSON in the uncompressed case, since we're just writing a buffer to a file and not using the whole `fs.writeJson(filename, data, { spaces: 2})`. 

I think the new behavior is okay. We don't need human-readable tablet files. But wanted to call this out as a side-effect of this PR.

Reply:
fyi the uncompressed file under `fs.writeFile` would look something like this (i tried it on s3):

```
{"version":"2","toolVersion":"0.0.0","snippets":{"a3fb238b791f212f2de9bf87c1625d6906fbdaac222f986c0a53ee5c652ec0b0":{"translations":{"python":{"source":"bucket = s3.Bucket(self, \"MyFirstBucket\")","version":"2"},"csharp":{"source":"Bucket bucket = new Bucket(this, \"MyFirstBucket\");","version":"1"},"java":{"source":"Bucket bucket = new Bucket(this, \"MyFirstBucket\");","version":"1"},"go":{"source":"bucket := s3.NewBucket(this, jsii.String(\"MyFirstBucket\"))","version":"1"},"$":{"source":"const bucket = new s3.Bucket(this, 'MyFirstBucket');","version":"0"}},"location":{"api":{"api":"moduleReadme","moduleFqn":"@aws-cdk/aws-s3"},"field":{"field":"markdown",
```